src/Rings/MPolyQuo.jl

```diff
@@ -498,13 +518,7 @@ true
 function Base.issubset(a::MPolyQuoIdeal{T}, b::MPolyQuoIdeal{T}) where T
   base_ring(a) == base_ring(b) || error("base rings must match")
   simplify!(a)
-  if !(isdefined(b, :SI))
-    simplify!(b)
-  end
-  if b.SI.isGB == false
-    b.SI      = Singular.std(b.SI)
-    b.SI.isGB = true
-  end
+  singular_groebner_assure!(b)
```

Comment:
Instead of always computing a GB, how about trying a `Singular.reduce` first without it and only compute a GB if the result is non-zero?

Reply:
I actually did have that though, but it is going to be more code.

Reply:
I think it'll be worth it. There are two types of ideals: 
(a) Ideals for which a GB computation is more involved.
(b) Ideals for which a GB computation terminates quickly, 

For (a), computing a reduction beforehand won't break the camel's back. For (b), the reduction beforehand should be instantaneous.

Reply:
What function do we use in Singular.jl for reduction by a non-GB ideal?